packages/liveblocks-node/src/client.ts

```diff
@@ -520,6 +558,33 @@ export class Liveblocks {
     };
   }
 
+  /**
+   * Like .getRooms(), but will return a continuous stream of all rooms that
+   * match the criteria.
+   */
+  // TODO Consider making this API public and supported?
+  async *#iterRooms(
+    criteria: RoomQueryCriteria,
+    options?: RequestOptions & { pageSize?: number }
+  ): AsyncGenerator<RoomData> {
+    const { pageSize, signal } = options ?? {};
+
+    let cursor: string | undefined = undefined;
+    while (true) {
+      const { nextCursor, data } = await this.getRooms(
+        { ...criteria, startingAfter: cursor, limit: pageSize },
+        { signal }
+      );
+      for (const room of data) {
+        yield room;
+      }
+      if (!nextCursor) {
+        break;
+      }
+      cursor = nextCursor;
+    }
+  }
```

Comment:
This is an abstraction over calling `.getRooms()` without having to think about pagination. I kept it private for now, but we can consider making it a public/supported API. What do you think @nimeshnayaju @CTNicholas ?

Reply:
Cool idea, I can see this being helpful.

Reply:
Love generator functions and is perfect for this use case. 

I also had a thought - if it made sense to use `yield` (generator functions) instead of the `flush` API. I am not suggesting we do that but @nvie have you considered that or thought of such alternative?

Reply:
@nimeshnayaju Not super sure what you mean. Do you mean to accept a coroutine instead of a regular callback function as the 2nd argument, so users could "yield" back to trigger a flush? I think that's not obvious enough to most users. Also, `flush()` no longer exists anymore. Flushing happens automatically now, you don't need to control it explicitly anymore.

Reply:
@CTNicholas OK, then I'll think I'll make this `.iterRooms()` helper public too as part of this change 👍

Reply:
@nvie I meant an alternative API where you could `yield` to `flush` changes, but I don't think it would have the same ergonomics that we now have with the automatic flush.

Reply:
Made the `.iterRooms()` API public/supported in b7ad1958f4a3dedba12486fe96f8a34d9d88a8e2.